x-pack/plugins/maps/public/classes/styles/vector/components/symbol/icon_preview.tsx

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Component } from 'react';
+import {
+  EuiColorPicker,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiPanel,
+  EuiSpacer,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import { mapboxgl, Map as MapboxMap } from '@kbn/mapbox-gl';
+import { i18n } from '@kbn/i18n';
+import { ResizeChecker } from '.././../../../../../../../../src/plugins/kibana_utils/public';
+import {
+  CUSTOM_ICON_PIXEL_RATIO,
+  createSdfIcon,
+  // @ts-expect-error
+} from '../../symbol_utils';
+
+export interface Props {
+  svg: string;
+  cutoff: number;
+  radius: number;
+  isSvgInvalid: boolean;
+}
+
+interface State {
+  map: MapboxMap | null;
+  iconColor: string;
+}
+
+export class IconPreview extends Component<Props, State> {
+  static iconId = `iconPreview`;
+  private _checker?: ResizeChecker;
+  private _isMounted = false;
+  private _containerRef: HTMLDivElement | null = null;
+
+  state: State = {
+    map: null,
+    iconColor: '#E7664C',
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this._initializeMap();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.svg !== prevProps.svg ||
+      this.props.cutoff !== prevProps.cutoff ||
+      this.props.radius !== prevProps.radius
+    ) {
+      this._syncImageToMap();
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    if (this._checker) {
+      this._checker.destroy();
+    }
+    if (this.state.map) {
+      this.state.map.remove();
+      this.state.map = null;
+    }
+  }
+
+  _setIconColor = (iconColor: string) => {
+    this.setState({ iconColor }, () => {
+      this._syncPaintPropertiesToMap();
+    });
+  };
+
+  _setContainerRef = (element: HTMLDivElement) => {
+    this._containerRef = element;
+  };
+
+  async _syncImageToMap() {
```

Comment:
remove async, method is not async and nothing awaits it

Reply:
thanks, I missed that.

Can you add check for `this._isMounted` after await. componentWillUnmount could have run before await returns and map be null destroyed at this point.

Reply:
Still need to add check for this._isMounted after await